core/src/main/java/org/apache/carbondata/core/datamap/DataMapStoreManager.java

```diff
@@ -92,6 +95,30 @@ private DataMapStoreManager() {
     return dataMaps;
   }
 
+  /**
+   * It only gives the visible datamaps
+   */
+  private List<TableDataMap> getAllVisibleDataMap(CarbonTable carbonTable) throws IOException {
+    CarbonSessionInfo sessionInfo = ThreadLocalSessionInfo.getCarbonSessionInfo();
+    List<TableDataMap> allDataMaps = getAllDataMap(carbonTable);
+    Iterator<TableDataMap> dataMapIterator = allDataMaps.iterator();
+    while (dataMapIterator.hasNext()) {
+      TableDataMap dataMap = dataMapIterator.next();
+      String dbName = carbonTable.getDatabaseName();
+      String tableName = carbonTable.getTableName();
+      String dmName = dataMap.getDataMapSchema().getDataMapName();
+      boolean isDmVisible = sessionInfo.getSessionParams().getProperty(
+          String.format("%s%s.%s.%s", CarbonCommonConstants.CARBON_DATAMAP_VISIBLE,
```

Comment:
Are you intend to make FG Datamap invisible? If yes, better to persistent this state by setting DataMapStatus in DataMapStatusDetail.

Reply:
And I think it is better to provide a statement for Disabling the datamap instead of a set command

Reply:
It's not going to disable the datamap forever. It is only intend to see how this datamap will enhance performance during the query.
As described in PR description, "Invisible datamap will only be ignored during query and user can still see this datmap and its data will be updated as normal."

It is named after MySQL document, and have the similar function as it. Please refer "MySQL now supports invisible indexes" in this link: https://dev.mysql.com/doc/refman/8.0/en/mysql-nutshell.html

Reply:
ok, so you mean this is temporary only? Not needed to persist it, right?

Reply:
Having a statement to do this means that we need to store the related information.

I also have considered to  store the visible/invisible information in DataMapSchema/DataMapStatus related, but found that currently there is no place other than file to store them.

Besides, this feature is introduced for tuning, which means the status may change frequently. Having a file stored information may need frequently refresh & read the DataMapSchemaFile.

By using a set command here, we don't have to make the information durable. It meets our needs and can be implemented easily.

@jackylk How do you think about it?

Reply:
yes, it is temporary, no need to persist it.

Reply:
I think it is ok, please add a @InterfaceStability.Unstable to the property key you have introduced

Reply:
@xuchuanyin It is applicable for all type of data maps like pre aggregate, timeseries, etc??

Reply:
@kumarvishal09 It
1. won't work on carbondata default datamap, such as DefaultBlockletDataMap/PreAggregate/Timeseries, 
2. and will work for extened datamaps, such as LuceneDataMap/BloomDataMap

The reason is that the visibility of datamap is done during blocklet pruning, it seems  PreAggregate/Timeseries won't arrive there and DefaultBlockletDataMap will always come to effective.